cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -261,37 +265,40 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
     TimedExaminationPreferenceEditor = AbstractEditor.extend({
         templateName: 'timed-examination-preference-editor',
         className: 'edit-settings-timed-examination',
-
         events : {
-            'change #id_timed_examination': 'timedExamination',
+            'change #id_not_timed': 'notTimedExam',
+            'change #id_timed_exam': 'timedExam',
+            'change #id_practice_exam': 'practiceExam',
+            'change #id_proctored_exam': 'proctoredExam',
             'focusout #id_time_limit': 'timeLimitFocusout'
         },
+        notTimedExam: function (event) {
+            event.preventDefault();
+            this.$('#id_time_limit_div').hide();
+            this.$('#id_time_limit').val('00:00');
+        },
+        timedExam: function (event) {
+            event.preventDefault();
```

Comment:
These 3 lines are copied two times below-- create a helper method (for example, "showTimeLimit").